scripts/model_gen_ng/model_gen_ng/struct.py

```diff
@@ -0,0 +1,188 @@
+import re
+from collections import defaultdict
+from typing import Dict, Iterable, List
+
+from . import DEFAULT_STRUCT_NAMESPACE, DEFAULT_USER_NAMESPACE
+from .definition import Definition
+from .jinja_utils import environment
+
+struct_template = environment.get_template("struct.tpl")
+struct_late_template = environment.get_template("struct_late.tpl")
+struct_forward_decls_template = environment.get_template("struct_forward_decls.tpl")
+class_forward_decls_template = environment.get_template("class_forward_decls.tpl")
+
+
+class StructField:
```

Comment:
There is quite a bit of code for this. I wonder how much we could avoid this by going either schemaless and use `struct["name"]` from jinja or autogenerate these classes from the JSON schema.

Not saying we need to do this now, but this quite a bit of boilerplate code that needs to stay in sync with the schema.

Reply:
Yeah the boilerplate classes are a bit tedious to write, but the templates are already not  so readable, adding `["propname"]` to each access would only worsen things. Besides, the classes also compute various properties (again, to avoid having logic in the template), and that will need to stay somewhere anyway...